options/options_test.cc

```diff
@@ -936,6 +938,46 @@ TEST_F(OptionsTest, GetBlockBasedTableOptionsFromString) {
   EXPECT_EQ(bfp->GetWholeBitsPerKey(), 4);
   EXPECT_EQ(bfp->GetMode(), BloomFilterPolicy::kAutoBloom);
 
+  // use_block_based_builder=true now ignored in public API (same as false)
+  ASSERT_OK(GetBlockBasedTableOptionsFromString(
+      config_options, table_opt, "filter_policy=bloomfilter:4:true", &new_opt));
+  bfp = dynamic_cast<const BloomFilterPolicy*>(new_opt.filter_policy.get());
```

Comment:
Should there be an assert to validate the cast worked?

Reply:
If the case is invalid, there will be a null deref, which for us reliably crashes in a discernible way. We always compile unit tests with RTTI.